src/main/java/org/folio/linked/data/service/impl/ResourceServiceImpl.java

```diff
@@ -75,13 +77,34 @@ public ResourceResponseDto createResource(ResourceRequestDto resourceDto) {
   }
 
   @Override
-  public Long createResource(org.folio.ld.dictionary.model.Resource modelResource) {
+  public Long saveMarcResource(org.folio.ld.dictionary.model.Resource modelResource) {
     var mapped = resourceModelMapper.toEntity(modelResource);
-    var persisted = saveMergingGraph(mapped);
-    refreshWork(persisted);
-    log.info("createResource [{}]\nfrom modelResource [{}]", persisted, modelResource);
-    applicationEventPublisher.publishEvent(new ResourceCreatedEvent(persisted));
-    return persisted.getId();
+    if (resourceRepo.existsById(modelResource.getId())) {
+      return saveRefreshAndPublishMarcResource(modelResource, mapped, ResourceUpdatedEvent::new);
+    }
+    var existedByInventoryId = ofNullable(modelResource.getInstanceMetadata())
+      .map(InstanceMetadata::getInventoryId)
+      .flatMap(resourceRepo::findByInstanceMetadataInventoryId);
+    if (existedByInventoryId.isPresent()) {
+      var existed = new Resource(existedByInventoryId.get());
+      return saveRefreshAndPublishMarcResource(modelResource, mapped,
+        saved -> new ResourceReplacedEvent(existed, saved));
+    }
+    return saveRefreshAndPublishMarcResource(modelResource, mapped, ResourceCreatedEvent::new);
+  }
+
+  private Long saveRefreshAndPublishMarcResource(org.folio.ld.dictionary.model.Resource modelResource, Resource mapped,
+                                                 Function<Resource, ResourceEvent> resourceEventSupplier) {
+    var newResource = saveMergingGraph(mapped);
+    refreshWork(newResource);
+    var event = resourceEventSupplier.apply(newResource);
+    if (event instanceof ResourceReplacedEvent rre) {
```

Comment:
Instead of doing an "instanceof ResourceReplacedEvent" check here, it will be better to call `breakEdgesAndDelete(rre.previous());` from line 89.

Reply:
I've made like that earlier, but there is a problem:
we have to refreshWork (line 99) for a newly saved resource before we break the connections and delete previous version, otherwise the MODLD-445 returns